rspace/src/main/scala/coop/rchain/rspace/Hash.scala

```diff
@@ -0,0 +1,18 @@
+package coop.rchain.rspace
+
+import coop.rchain.crypto.hash.Blake2b256
+
+class Hash private (val bytes: Array[Byte])
```

Comment:
This information about hashing function used gets lost when creating the instance of `Hash`. If we always use the `Blake2b256` hashing function that's probably fine but if we add other functions in the future we will have to assume that all hashes created up to that point were created using `Blake2b256`. If you think that this approach is OK for now than I am good.

Reply:
Can you add a comment why you chose to represent it as a `class` rather than a `case class`? I guess that it's to have your own serialization.

Reply:
I will add a comment.

For reference here:  it allows me to only make public a smart constructor to ensure that nobody is taking an arbitrary byte array and shoving it into a `Hash`